modules/search/class.jetpack-search-widget-filters.php

```diff
@@ -140,10 +149,94 @@ function widget( $args, $instance ) {
 			do_action( 'jetpack_search_render_filters_widget_title', $title, $args['before_title'], $args['after_title'] );
 		}
 
+		// we need to dynamically inject the sort field into the search box when the search box is enabled, and display
+		// it separately when it's not.
 		if ( ! empty( $instance['search_box_enabled'] ) ) {
 			$this->render_widget_search_form( $instance );
 		}
 
+		$default_sort = isset( $instance['sort'] ) ? $instance['sort'] : self::DEFAULT_SORT;
+		list( $orderby, $order ) = $this->sorting_to_wp_query_param( $default_sort );
+		$current_sort = "{$orderby}|{$order}";
+
+		if ( ! empty( $instance['search_box_enabled'] ) && ! empty( $instance['user_sort_enabled'] ) ) {
+			?>
+			<label class="jetpack-search-sort-wrapper">
+				<?php esc_html_e('Sort by', 'jetpack'); ?>
+				<select name="<?php echo esc_attr( $this->get_field_name( 'sort' ) ); ?>" class="jetpack-search-sort">
+					<?php foreach( $this->get_sort_types() as $sort => $label ) { ?>
+						<option value="<?php echo $sort; ?>" <?php selected( $current_sort, $sort ); ?>>
+							<?php echo $label; ?>
+						</option>
+					<?php } ?>
+				</select>
+			</label> <?php
+		}
+
+		/*
+		 * this JS is a bit complicated, but here's what it's trying to do:
+		 * - find or create a search form
+		 * - find or create the orderby/order fields with default values
+		 * - detect changes to the sort field, if it exists, and use it to set the order field values
+		 */
+		?>
+		<script type="text/javascript">
+			jQuery( document ).ready( function( $ ) {
+				var actionUrl      = <?php echo json_encode( home_url( '/' ) ); ?>,
+					orderByDefault = <?php echo json_encode( $orderby ); ?>,
+					orderDefault   = <?php echo json_encode( $order ); ?>,
+					widgetId       = <?php echo json_encode( $this->id ); ?>,
+					currentSearch  = <?php echo json_encode( isset( $_GET['s'] ) ? $_GET['s'] : '' ); ?>
+
+				var container = $('#' + widgetId);
```

Comment:
Since the container is defined by the `before_widget` argument in `register_sidebar()`, which is defined by the theme, we're not guaranteed to always have `#{ $widget_id }`. The default is `'<li id="%1$s" class="widget %2$s">'`, but a theme would register a `before_widget` argument without the ID.

Perhaps we should add a container around our whole widget that was something like `<div id="jetpack-search-widget_$id>`?

Reply:
While it's theoretically possible for a theme to register a widget without the widget's ID as the `id` property, I haven't actually seen a single theme that does that.

I am torn. Part of me feels like it would break markup more to have an extra DIV inside the widget, but maybe it's totally safe... 

Alternatively we could scrape the `id` value out of `$before_widget`? That feels brittle and overly complicated.

Reply:
> While it's theoretically possible for a theme to register a widget without the widget's ID as the id property, I haven't actually seen a single theme that does that.

For what it's worth, I've also never seen it, but ¯\_(ツ)_/¯.

> Part of me feels like it would break markup more to have an extra DIV inside the widget, but maybe it's totally safe...

I don't think it would cause an issue, but I could be wrong.

> Alternatively we could scrape the id value out of $before_widget? That feels brittle and overly complicated.

Theoretically, the theme could just omit the ID argument. I think our best bests are to either 1) decide not to address this or 2) use a wrapper div where we controls the id or class.